src/main/drivers/dma.c

```diff
@@ -89,13 +89,11 @@ uint32_t dmaFlag_IT_TCIF(const dmaResource_t *channel)
 }
 
 #define DMA_RCC(x) ((x) == DMA1 ? RCC_AHBPeriph_DMA1 : RCC_AHBPeriph_DMA2)
-void dmaInit(dmaIdentifier_e identifier, resourceOwner_e owner, uint8_t resourceIndex)
+dmaEnable(dmaIdentifier_e identifier) {
```

Comment:
Do we also need a `dmaDisable()` in order to be able to back out if we get the Rx stream and not the Tx stream, or perhaps we should just do more checks before calling `dmaEnable()`?

Reply:
It wasn't needed in the use cases that I've come across.

My thinking would be to rather add a `dmaFree()` that undoes the reservation that `dmaAllocate()` does - this way drivers can try to allocate all DMA resources, and bail out before starting the initialisation of the hardware.

But `dmaDisable()` might be needed for other cases as well.